test/exceed_test.gd

```diff
@@ -0,0 +1,279 @@
+				# There is sometimes an init_extra_cost here
+				# TODO: See if anyone needs it to be more than 0
+				assert_true(game_logic.do_pay_strike_cost(player, content, false),
+						"%s failed to pay a Strike cost using %s" % [player, content])
+			var decision_type:  # Unknown decision type, just roll with it
+				assert_true(game_logic.do_choice(player, content),
+						"Decision of type %s unhandled by test harness (attempted by player %s, content %s)" % [
+								decision_type, player, content])
+
+func process_remaining_decisions(initiator, defender, init_choices, def_choices):
+	var empty_loop_count = 0
+	while init_choices.size() + def_choices.size() >= 1:
+		# empty_loop_count is used to detect when there are still prescribed choices in
+		# the input, but the game is not actually making additional decision points available.
+		empty_loop_count += 1
+		if empty_loop_count >= 3:
+			fail_test("Game is not providing decision points to process initiator" +
+					" choices %s or defender choices %s" % [init_choices, def_choices])
+			return
+		while game_logic.game_state == Enums.GameState.GameState_PlayerDecision:
+				## TODO: Figure out if it's really necessary to limit ourselves to this decision type
+				# and game_logic.decision_info.type == Enums.DecisionType.DecisionType_ChooseSimultaneousEffect:
+			empty_loop_count = 0  # reset the count each time we actually get into this loop; it is not empty
+			var decision = game_logic.decision_info
+			var player = initiator
+			var player_choices = init_choices
+			if decision.player == defender.my_id:
+				player = defender
+				player_choices = def_choices
+			var choice = player_choices.pop_front()
+			if choice == null:
+				fail_test("Insufficient decisions defined for player %s during strike" % player.my_id)
+				return
+			match game_logic.decision_info.type:
+				Enums.DecisionType.DecisionType_ChooseSimultaneousEffect:
+					assert_true(game_logic.do_choice(player, choice),
+							"%s failed to perform a choice with value %s" % [player, choice])
+				Enums.DecisionType.DecisionType_ChooseToDiscard:
+					assert_true(game_logic.do_choose_to_discard(player, choice),
+							"%s failed to discard cards %s" % [player, choice])
+				Enums.DecisionType.DecisionType_ForceForArmor:
+					assert_true(game_logic.do_force_for_armor(player, choice),
+							"%s failed to discard cards %s for armor" % [player, choice])
+				var decision_type:  # Unknown decision type, just roll with it?
+					if typeof(choice) == Variant.Type.TYPE_ARRAY:
+						fail_test("Attempting to apply array choice %s to a decision of type %s" % [
+								choice, Enums.DecisionType.keys()[decision_type]])
+						return
+					assert_true(game_logic.do_choice(player, choice),
+							"Decision of type %s unhandled by test harness (attempted by player %s, content %s)" % [
+									decision_type, player, choice])
+		## TODO: Does the loop need to wait a tick or two here for the game engine to
```

Comment:
there is nothing time based inside localgame

Reply:
See below.